### ISSUE TITLE
`WaTextarea.autocorrect` typed as `string` conflicts with TypeScript DOM lib (`boolean`)

### ISSUE THREAD

**Issue by MatiasPuhakka** [shoelace-style/webawesome#2299]
### Describe the bug

`WaTextarea` types the `autocorrect` property as `string`, but TypeScript's DOM lib (introduced in TypeScript 5.x) types `HTMLElement.autocorrect` as `boolean`. This causes a structural type incompatibility: `WaTextarea` no longer satisfies `extends HTMLElement` in TypeScript's strict type checker, breaking any generic code that constrains on `HTMLElement`.

```
TS2419: Types of construct signatures are incompatible.
  Type 'new () => WaTextarea' is not assignable to type 'new () => HTMLElement'.
    The types of 'autocorrect' are incompatible between these types.
      Type 'string' is not assignable to type 'boolean'.
```

### To Reproduce

1. Install WebAwesome and TypeScript ≥ 5.4 (where `HTMLElement.autocorrect` became `boolean` in the DOM lib)
2. Add this to any `.ts` file:
   ```ts
   import WaTextarea from '@awesome.me/webawesome/dist/components/textarea/textarea.js';

   const _: new () => HTMLElement = WaTextarea; // TS2419
   ```
3. Run `tsc --noEmit` — error is reported on that line

### Demo

Minimal reproduction in any TypeScript 5.4+ project with WebAwesome installed — no additional dependencies required.

### Screenshots

N/A — compile-time error only.

### Browser / OS

N/A — TypeScript type-checking error, not a runtime browser issue.

### Additional information

The fix is to change `WaTextarea`'s `autocorrect` property type from `string` to `boolean` (or `string | boolean` if backward compatibility is needed). The HTML spec defines `autocorrect` as an enumerated attribute mapped to a boolean IDL attribute in newer specs.

**Comment by claviska** [comment]
This was [fixed in 3.4.0](https://webawesome.com/docs/resources/changelog#wa_340) as part of [this commit](https://github.com/shoelace-style/webawesome/commit/af9a076d16f46c2d006d29a0ee5c97ccf86ccdcd#diff-83030450d668ca75b0756ad3583823707a0ca515b9727af0e76ef9d97856d76cR129-R140).

If I've missed something, feel free to comment here :)
